polling_stations/apps/addressbase/models.py

```diff
@@ -18,10 +28,50 @@ def points_for_postcode(self, postcode):
         qs = qs.values_list("location", flat=True)
         return list(qs)
 
+    def uprns_for_postcode(self, postcode):
+        postcode = Postcode(postcode)
+        uprns = []
+        addresses = Address.objects.filter(postcode=postcode.with_space)
+
+        for a in addresses:
+            uprn_info = UprnInfo(
```

Comment:
2. I think you can now totally bin `UprnInfo` now here. An object that has the properties

```
.address
.postcode
.council(_id)
.uprn
.polling_station_id
.location
```

is now just an `Address` object :) You don't need to wrap it in a NamedTuple

That said, if you've got a postcode and you want all the Address objects in that postcode, remember the `AddressBaseGeocoder` object has a `.addresses` property https://democracyclub.github.io/uk-geo-utils/geocoders/#addresses that does several useful things:

- Returns all the Address objects with that postcode (duh)
- Uses a cache to reduce Database IO if possible
- Attempts to sort the addresses into a human-useful order (e.g: for use in an address picker, or a dashboard)

so if you've got an `AddressBaseGeocoder` object in scope when you need this it is more efficient to use that if possible and if you're making a list of addresses to be read by a human, its probably worth using it for the sorting.

Reply:
OK, I left the UprnInfo  in because I didn't want to dive back into the dashboard code - I tend to get pretty confused in there - but it wasn't too bad :). 

I think the places we get addresses by postcode is in the AddressFormView and the RoutingHelper. The former just hit the database similar to now (different tables), but the latter might benefit from the `addresses` property being a cached property?

Reply:
Also I don't think either of the `AddressManager` methods are getting used anywhere... so perhaps the whole class can go.